### ISSUE TITLE
When select noindex or nofollow write rules to robots.txt file

### ISSUE THREAD

**Issue by tonyzeoli** [semperfiwebdesign/all-in-one-seo-pack#2637]
**Is your feature request related to a problem? Please describe.**

When I click on a checkbox to default NO INDEX (or NO FOLLOW) a category or a post type, I would think the rule should be written to the robots.txt file editor, so that a site admin can see the rule that was put in place.

Recently, Google reported I had some categories set as NO INDEX. So, I checked the Robots.txt editor in AIOSEO to see if that was the case and did not see that rule, so I didn't know if or how I was blocking the categories reported. If the rule is not in Robots.txt, then how can I ascertain what Google is warning me about?

**Describe the solution you'd like**

When I click a checkbox in AIOSEO to set a rule to default NO INDEX or NO FOLLOW, the rule should be displayed in the robots.txt file for the site (I'm running multisite).

**Additional context**

**Comment by michaeltorbert** [comment]
Hi Tony,

Offhand I'd say this wouldn't be good behavior, because they're two separate things.
For instance, you may have noindex but follow, and so you wouldn't want to blog Google from crawling the page.
Also, if there are a lot of categories, that's a LONG robots.txt. :)
This is also something that introduces another layer of complexity (and bugs) for redundant functionality. 

If I understand correctly, you're wanting it so a site admin can at a glance see the status, but can't they just go to the AIOSEOP settings page and see what they want?

**Comment by tonyzeoli** [comment]
Consider the following.

Google sends an email to the site admin reporting that something is blocked
in Robots.txt. The site admin is new to the company or someone else may
have changed a setting when they had access to it. The site admin reviews
Robots.txt and sees nothing. The site admin is new to AIOSEO, so he/she
doesn't know what to do next, so they send a support request to AIOSEO to
ask, because there's no specific error displaying anywhere or some
notification that tells them something has been added to Robots.txt, but is
not visible. That ends up taking everyone's time when there could be some
notification, somewhere - maybe the Dashboard.

There should be some way to indicate to the user that NO INDEX DEFAULT(S)
are active, so at least when they check the plugin, they can say, "oh, I
see...there must be a default setting somewhere, the click the link to it
and can then see what is incorrectly set.

Also, maybe there can be a note in the Robots.txt feature that says
something like:

"Checking a setting in AIOSEO will not write the rule to this Robots.tx
file. If you receive an email from Google stating that a post type,
category, or taxonony is blocked by Robots.txt, please check your settings,
fix, and then resubmit the errors in the Google console."

It would be helpful to remind users just like WordPress reminds users that
there is a setting under General > Reading that is blocking search engines,
that they've blocked some things by default and that they should address it
or it will have a negative impact.

It's just a helpful reminder to anyone who is new to AIOSEO or who knows
the plugin but just needs a nudge to remind them they've changed a setting.
Because the first place I looked wasn't the settings panel. It was
Robots.txt to see what was blocked and why. When I saw nothing, I then got
confused and reached out to support. At that point, if I had information, I
wouldn't have sent a support request and could have solved the problem
myself.

**Comment by michaeltorbert** [comment]
Is Google sending emails that the robots.txt is blocking access when it isn't, it's really the robots meta tag on the page itself? I've never seen this ad we've never gotten any reports about it as far as I know.
@wpsmort can you weigh in here?

**Comment by tonyzeoli** [comment]
Here's the text of the message from Google Search Console below.

So, when the URL is marked "no index," the obvious place to look first is
Robots.txt, because I wanted to see if the URL was actually marked
"noindex." When I looked at the Errors in Search Console, it showed me the
three URLs and I ascertained they were category URLs. It suggested I
inspect the URLs, so I inspected them and they all passed with a green
"allow" in the old search console, because the tool hasn't moved to the new
search console yet. Once I ran them through the inspection and found that
they were all "allowed," I then thought - "well, how could they be allowed
if Google is telling me they are noindex. So my next step was to look in
Robots.txt to see if the URLs were there and they weren't. Then, I got
stuck and thought, "what do I do next here?" Why can't I see these URLs as
being "noindex." What do I check next? Because it didn't occur to me that
the NOINDEX for Categories setting was enabled. I then went to WordPress
SEO and Internet Questions group and posed the question about why this I
wasn't seeing these URLs in Robots.txt and it wasn't until I upgraded to
3.0 that I clearly saw the category setting was checked.

If URLs can't be printed to Robot.txt for a variety of reasons, at least it
makes sense to display a notification that say, "You have set X, Y, and Z
setting to NOINDEX" as either a notification that can be closed to remind
the person that if they get this info from Google, that there is a good
reason why they are receiving the message and here's how to solve it. It's
just a UX helper to give the user a prompt to remember that decisions were
made and that any time you select one of these settings, you're not going
to see them passed into Robots.txt, which is the first obvious place you'd
look to ascertain if a URL had a NOINDEX or NOFOLLOW on it.

I guess all I'm looking for is a notification that says that I've made some
decisions that could affect the visibility of pages, posts, categories,
taxonomies and remind me that they've been checked, so when Google tells me
something, A. I am reminded I won't see those URLs in Robots.tx, and B. to
check those panels to view my settings as that could be the cause of my
issue.

Does that make sense?

New Coverage issue detected for site 67biltmore.com

To owner of 67biltmore.com,

Search Console has identified that your site is affected by 1 new Coverage
related issue. This means that Coverage may be negatively affected in
Google Search
results. We encourage you to fix this issue.

New issue found:

Submitted URL marked ‘noindex’


[image: photo]
Tony Zeoli
Founder, Digital Strategy Works LLC
O  (828) 412-0990  <(828) 412-0990> M  (917) 705-4700  <(917)
705-4700> E  az@digitalstrategyworks.com
<az@digitalstrategyworks.com>
W  https://digitalstrategyworks.com
Whatsapp 9177054700
Skype tonyzeoli
<http://www.facebook.com/digitalstrategyworks>
<http://www.twitter.com/dswks>
<http://www.linkedin.com/company/digital-strategy-works>
<http://plus.google.com/digitalstrategyworks>
<http://slideshare.com/tonyzeoli> <http://about.me/tonyzeoli>
<http://angel.co/tonyzeoli>
<https://calendly.com/tonyzeoli> Schedule a 30-minute Meeting
<https://calendly.com/tonyzeoli>
<https://digitalstrategyworks.com/drupal-to-wordpress-migration-for-bookstr-com/2019/02/05/2575/>
Latest
post: Drupal to WordPress Migration for Bookstr.com
<https://digitalstrategyworks.com/drupal-to-wordpress-migration-for-bookstr-com/2019/02/05/2575/>
<https://facebook.com/digitalstrategyworks>   <https://twitter.com/dswks>
<https://linkedin.com/in/+tonyzeoli>
<https://www.youtube.com/+digitalstrategyworks>


<https://gsuite.google.com/landing/partners/referral/trial.html?utm_source=sign-up&utm_medium=referralbutton1&utm_campaign=apps-referral-program&utm_content=https://goo.gl/BG4yH8>

**Comment by wpsmort** [comment]
The messages from GSC are very specific.  If it says "Submitted URL marked ‘noindex’" then this is caused by the robots meta tag in the source code.  Whereas, if it says "Submitted URL blocked by robots.txt" then it's because the URL is blocked in the robots.txt file.  These are two separate things and should not be confused.  Google have a list of these errors and explanations here - https://support.google.com/webmasters/answer/7440203#error-status

I agree with Michael, we definitely don't want to add content marked as noindex to the robots.txt for the reason he specified.  

There are three things we could do to help users out: 

1. In the Robots.txt module, we should add a heading above the window that displays the robots.txt to indicate that it's displaying the robots.txt - see #1127 
2. We could, as Tony suggests, have a dismissible notice on the notification dashboard (#998) to indicate when post types or taxonomies are set to noindex.  I personally would prefer a notice that reminds users to periodically review their Noindex Settings and their Robots.txt.  
3. We could add FAQ's to our docs that explains what Noindex is and what Robots.txt is.

Let me know your thoughts.  If you agree with 2 and 3 above then I'll open separate issues for those.

**Comment by tonyzeoli** [comment]
>
> I definitely think your solutions are on target and what I would expect to
> see. For the novice SEO, they might not understand the distinctions or even
> look for them based on what Google reports. While I'm somewhat skilled and
> knowledgeable, I didn't know their are two distinctions Google might
> report, because I only got the one message and never got the other you
> referenced, so how would I know the difference unless I received both of
> them and saw that Google was making a distinction.


When you're (the developers) as close to this as anyone, it might seem
obvious to you these distinctions. But it's not obvious until someone tells
you it exists. Therefore, these helper guides in your docs and in the
plugin would certainly be helpful to indicate settings have been changed
and here's where to look.

Lastly, as i said before, someone may walk into this for the first time and
have no idea what to look for, so Steve, you're suggestions are on point
and it would be nice to see them in the plugin. Thanks.

>
>

**Comment by wpsmort** [comment]
I'm closing this in favor of the issues opened as per my comment here https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/2637#issuecomment-501074057

**Comment by tonyzeoli** [comment]
Thanks, Steve!
